tests/test_transactions.py

```diff
@@ -146,3 +147,22 @@ def test_transaction_bad_repo(fs: LakeFSFileSystem) -> None:
     with pytest.raises(ValueError, match="repository .* does not exist"):
         with fs.transaction(repository="REEEE"):
             pass
+
+
+def test_warn_uncommitted_changes(
+    random_file_factory: RandomFileFactory,
+    fs: LakeFSFileSystem,
+    repository: Repository,
+    temp_branch: Branch,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    random_file = random_file_factory.make()
+
+    lpath = str(random_file)
+
+    with caplog.at_level(logging.WARNING):
+        with fs.transaction(repository, temp_branch) as tx:
+            fs.put_file(lpath, f"{repository.id}/{tx.branch.id}/{random_file.name}")
+        assert len(caplog.records) == 1
+        assert caplog.records[0].levelname == "WARNING"
+        assert "uncommitted changes" in caplog.text
```

Comment:
Switch this to a `with pytest.warns(...)` with an appropriate regex. You might have to explicitly test `delete = "never"` to make the codecov bot happy.